src/store/mod.rs

```diff
@@ -134,40 +130,45 @@ pub struct StoreConfig {
     /// optional, and unused internally.
     pub size: Option<usize>,
 
-    /// The number of merkle tree levels above the base to cache on disk.
-    pub levels: usize,
+    /// The number of merkle tree rows_to_discard then cache on disk.
+    pub rows_to_discard: usize,
 }
 
 impl StoreConfig {
-    pub fn new<T: Into<PathBuf>, S: Into<String>>(path: T, id: S, levels: usize) -> Self {
+    pub fn new<T: Into<PathBuf>, S: Into<String>>(path: T, id: S, rows_to_discard: usize) -> Self {
         StoreConfig {
             path: path.into(),
             id: id.into(),
             size: None,
-            levels,
+            rows_to_discard,
         }
     }
 
     // If the tree is large enough to use the default, use it.  If
     // it's too small to cache anything, don't cache anything.
-    // Otherwise, the tree is 'small' so a fixed value of 2 levels
-    // above the base should be sufficient.
-    pub fn default_cached_above_base_layer(leafs: usize, branches: usize) -> usize {
+    // Otherwise, the tree is 'small' so a fixed value of height - 2
+    // rows_to_discard should be sufficient.
+    pub fn default_rows_to_discard(leafs: usize, branches: usize) -> usize {
         let height = get_merkle_tree_height(leafs, branches);
         if height <= 2 {
+            // If a tree only has a root row and/or base, there is
+            // nothing to discard.
             return 0;
+        } else if height == 3 {
+            // If a tree only has 1 row between the base and root,
+            // it's all that can be discarded.
```

Comment:
As above, I think you are either still off by one — or else using the wrong definition of height.

Reply:
Wrong definition of height